lib/bomb.js

```diff
@@ -56,7 +57,7 @@ var bomb = {
      */
     trim: function (str) {
         var pos = bomb.hasBOM(str);
-        return pos ? str.splice(post) : str;
+        return typeof str === 'string' && typeof str.slice === 'function' && str.slice(pos) || str;
```

Comment:
no need to do such extensive check. if pos is truthy str must have been a string!